teek/_widgets/treeview.py

```diff
@@ -0,0 +1,452 @@
+                                    '-' + option, *args)
+
+    @make_thread_safe
+    def assign(self, treeview):
+        # Assign row to given treeview
+        self._treeview = treeview
+
+    @make_thread_safe
+    def to_tcl(self):
+        # Represented by name
+        return self._name
+
+    @property
+    def selected(self):
+        """
+        Get current selection state
+        """
+        return self._name in self._treeview._call([str], self._treeview,
+                                                  'selection')
+
+    @make_thread_safe
+    def select(self):
+        """
+        Add row to current selection
+        """
+        self._treeview._call(None, self._treeview, 'selection', 'add', self)
+
+    @make_thread_safe
+    def deselect(self):
+        """
+        Remove row from current selection
+        """
+        self._treeview._call(None, self._treeview, 'selection', 'remove', self)
+
+
+class TreeviewColumnList(MutableSequence):
+    """
+    List containing all columns of a treeview
+    """
+
+    def __init__(self, treeview):
+        super().__init__()
+
+        # Default column 0
+        self._data = [TreeviewColumn('#0', treeview=treeview)]
+        self._treeview = treeview
+
+    def __setitem__(self, index, column):
+        if index == 0:
+            raise KeyError('cannot set column #0')
+
+        self._data[index] = column
+        self._update()
+
+    def __delitem__(self, index):
+        if index == 0:
+            raise KeyError('cannot delete column #0')
+
+        del self._data[index]
+        self._update()
+
+    def __getitem__(self, index):
+        return self._data[index]
+
+    def __len__(self):
+        # Length includes column 0
+        return len(self._data)
+
+    def _update(self):
+        """
+        Update columns of widget and rebuild column objects
+        """
+        self._treeview._call(None, self._treeview, 'configure', '-columns',
+                             self._data[1:])
+
+        for column in self._data:
+            column.config.push()
+
+    def insert(self, index, column):
+        """
+        Insert column or column text converted to a column at given index.
+        Note that the column at index 0 cannot be replaced!
+        """
+        if index == 0:
+            raise KeyError('cannot insert column at index 0')
+
+        # Convert to column
+        if not isinstance(column, TreeviewColumn):
+            column = TreeviewColumn(text=column)
+
+        column.assign(self._treeview)
+        self._data.insert(index, column)
+        self._update()
+
+
+class TreeviewRowList(MutableSequence):
+    """
+    List containing all rows of a treeview
+    """
+
+    def __init__(self, treeview):
+        super().__init__()
+
+        self._data = []
+        self._treeview = treeview
+
+    def __getitem__(self, index):
+        return self._data[index]
+
+    def __setitem__(self, index, row):
+        raise NotImplementedError('use insert() instead')
```

Comment:
What's wrong with implementing this using `TreeviewRowList.insert` and `TreeviewRowList.delete`?

Reply:
I implemented the management of rows and columns like a list. Therefore, `.insert()` is available anyway. For deleting items, `del item` can be used. I think this is more pythonic than an additional `.delete()` method.